File: java/src/test/java/publicdomain/test/snippets/fixtures/SimulatedTimeFixture.java
package publicdomain.test.snippets.fixtures;

import publicdomain.test.snippets.system.SimulatedTime;

import java.time.Duration;

/**
 * {@link SimulatedTimeFixture} handles an instance of {@link SimulatedTime} as a test fixture.
 * <p>
 * Depending on how the system under test receives the reference to the {@link publicdomain.test.snippets.system.Time},
 * the fixture could provide a getter to retrieve the instance, and the test could inject it.
 */
public class SimulatedTimeFixture {
    private final SimulatedTime time = new SimulatedTime();

    public void passes(final Duration duration) {
        time.pass(duration);
    }
}

File: dotnet/TestSnippets.ForTest/Fixtures/SimulatedTimeFixture.cs
using System;
using PublicDomain.TestSnippets.System;

namespace PublicDomain.TestSnippets.Fixtures
{
    /// <summary>
    /// <see cref="SimulatedTimeFixture"/> handles an instance of <see cref="SimulatedTime"/> as a test fixture.
    /// 
    /// Depending on how the system under test receives the reference to the <see cref="ITime"/>,
    /// the fixture could provide a getter to retrieve the instance, and the test could inject it.
    /// </summary>
    public class SimulatedTimeFixture
    {
        private readonly SimulatedTime _time = new();

        public void Passes(TimeSpan span)
        {
            _time.Pass(span);
        }
    }
}